### ISSUE TITLE
Little issue with type inference for sorted built-in function

### ISSUE THREAD

**Issue by AlexanderPodorov** [microsoft/pyright#6384]
**Describe the bug**
See example code.

**Code or Screenshots**

```python
# This code does not pass strict type checking.
# Looks like due to the inferred type of (item: _T@sorted | int)
test = sorted([3] + [3.13, 3.14], key=lambda item: item)

# Identical code passes strict type checking.
# inferred type of item is "float | int", which is correct.
ll = [3] + [3.13, 3.14]
test = sorted(ll, key=lambda item: item)
```

**VS Code extension or command-line**
Pylance v2023.11.10
I believe Pyright version is 1.1.333

**Comment by erictraut** [comment]
Although this wasn't obvious, this turns out to be the same underlying issue as https://github.com/microsoft/pyright/issues/6273, which I fixed in the recently-released 1.1.337. So this bug no longer occurs with the new version of [pyright](https://pyright-play.net/?strict=true&code=MQAgKgFglgziDGB7AJgUxMxq4DtEBcQAHAQxjhnwCcp5D8BPI9eCVeAayhwHMA6AFCgAMokQc4AGygd0yAK7p8iEPjYhuAM1RUqqZKqbpEmkAAoo%2BVAFsAXCAD6YAAIxEVKwYA%2BGnPgCUAlaUIAC8IG4e%2BmYA2gDMALogANQg8XwAjHEANCBxmQAsCbmyDKGSJNYARsgkGlZ29TaBQiAAkmh%2BtCSSCCjopOTYEdS09EYIbJzc-K1aOnoGjMwgJk3WGnAARJqSiCSEPtz4W7kA7tCsm3267PiCkr3h8Ump6Vm5%2BRlFQdiE4ZFPGZHiVUGUKtVaut7JZmgIgA).
